presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/decoders/ArrayDecoder.java

```diff
@@ -50,7 +43,9 @@ else if (data instanceof List) {
             output.closeEntry();
         }
         else {
-            throw new PrestoException(TYPE_MISMATCH, format("Expected list of elements for field '%s' of type ARRAY: %s [%s]", path, data, data.getClass().getSimpleName()));
+            BlockBuilder array = output.beginBlockEntry();
+            elementDecoder.decode(hit, () -> data, array);
+            output.closeEntry();
```

Comment:
What type is observed here during your investigation?

what about unsuported type? why should we not have handling for that?

Reply:
getting error like Expected a string or numeric value for field 'songs' of type VARCHAR: [hello, hi, ala] [ArrayList]...now its fixed

Reply:
Unsupported types should give exception. handle this specific case and add else case which raises exception